### DISCUSSION TITLE
HTTP Error 500 upon trying to dispose devices

### DISCUSSION THREAD

**Discussion by gegeny** [opendcim/openDCIM#1504] [Q&A] [answered]
Hi,

I am facing an issue when trying to dispose a device. The device is in the storage room and I try to add it to an active disposal method. Upon clicking 'Go', openDCIM gives me an HTTP 500 message. The corresponding error log:
[Tue Jan 30 09:39:18.550544 2024] [php:error] [pid 23078] [client 10.1.1.197:61649] PHP Fatal error:  Uncaught PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'set DispositionID='6', DispositionDate=NOW(), DisposedBy='gegeny' where Devic...' at line 1 in /opt/openDCIM/openDCIM-23.04/classes/DispositionMembership.class.php:41\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/classes/DispositionMembership.class.php(41): PDOStatement->execute()\n#1 /opt/openDCIM/openDCIM-23.04/classes/Device.class.php(675): DispositionMembership->addDevice()\n#2 /opt/openDCIM/openDCIM-23.04/storageroom.php(24): Device->Dispose()\n#3 {main}\n  thrown in /opt/openDCIM/openDCIM-23.04/classes/DispositionMembership.class.php on line 41, referer: https://opendcim.ilonetwork.net/storageroom.php?dc=3

The issue can be reproduced in the Demo instance:
1. Move device to Storage Room
2. Try to add device from Storage Room to an active Disposal Method

OpneDCIM 23.04
MariaDB 10.6.15
PHP 8.0.30
Apache 2.4.51
SLES15 SP5

Thank you,
Zoltan

**Comment by samilliken** [comment, marked as answer]
Please look at close issues for duplicates before starting up a new thread.

https://github.com/opendcim/openDCIM/issues/1493

**Reply by gegeny** [reply]
Thank you, Scott!

I missed #1493. I confirm that the change fixed the issue.

Kind regards,
Zoltan
